src/compiler/parser.ts

```diff
@@ -1543,17 +1546,18 @@ namespace ts {
         }
 
         function parseExpectedMatchingBrackets(openKind: SyntaxKind, closeKind: SyntaxKind, openPosition: number) {
-            if (!parseExpected(closeKind)) {
-                const lastError = lastOrUndefined(parseDiagnostics);
-                if (lastError && lastError.code === Diagnostics._0_expected.code) {
-                    addRelatedInfo(
-                        lastError,
-                        createDetachedDiagnostic(fileName, openPosition, 1, Diagnostics.The_parser_expected_to_find_a_1_to_match_the_0_token_here, tokenToString(openKind), tokenToString(closeKind))
-                    );
-                }
-                return false;
+            if (token() === closeKind) {
+                nextToken();
+                return true;

```

Comment:
Looks like no part of the parser consumes the return value, so you can make this `void`-returning.

Reply:
Yeah, it looks like it was just matching the nearby boolean functions.